provider/datagen/src/lib.rs

```diff
@@ -91,6 +88,14 @@ pub mod syntax {
     pub use icu_provider_fs::export::serializers::postcard::Serializer as Postcard;
 }
 
+/// A prelude for using the datagen API
+pub mod prelude {
+    pub use super::{syntax, CldrLocaleSubset, CollationHanDatabase, Out, SourceData};
+    pub use icu_locid::{langid, LanguageIdentifier};
+    pub use icu_provider::KeyedDataMarker;
+}
```

Comment:
The prelude is a definite improvement.

Suggestion (optional): It still might not be ideal for things like `langid` to be imported, since it's easy to imagine it clashing with something else in the namespace. You can avoid this by just exporting the three crates, and then qualifying the call sites:

```rust
pub mod prelude {
    pub use crate as icu_datagen;
    pub use icu_locid;
    pub use icu_provider;
}
```

Reply:
`langid` is in the macro namespace, I don't think it's that easy to imagine the client also having a macro called `langid`.